value/mapunstructured.go

```diff
@@ -88,12 +88,12 @@ func (m mapUnstructuredInterface) EqualsUsing(a Allocator, other Map) bool {
 	}
 	vv := a.allocValueUnstructured()
 	defer a.Free(vv)
-	return other.Iterate(func(key string, value Value) bool {
+	return other.IterateUsing(a, func(key string, value Value) bool {
 		lhsVal, ok := m[key]
 		if !ok {
 			return false
 		}
-		return Equals(vv.reuse(lhsVal), value)
+		return EqualsUsing(a, vv.reuse(lhsVal), value)
```

Comment:
Here's we're doing `vv.reuse` when we've done `defer a.free(vv)` above, is that fine?

Reply:
OK I see why it's fine, we don't return the closure, we use it in IterateUsing, so once this function is done, we're good to go. cool.